### PR TITLE
registry: Implement URL paths and independent image views

### PR DESCRIPTION
**Pull request by stefwalter**
Also default activate elements by moving to their page. This behavior can be changed in the menu, but is this way for simplicity.
    
When only dealing with admins, we were able to build a tool that required them to learn how the UI worked and get used to it. When dealing with users, for the self-service image registry UI, we don't have that luxury. So keep things simple.

### CODE REVIEW

**Comment by stefwalter** [comment]
Depends on #3440

**Comment by stefwalter** [comment]
Fixed test failures under phantomjs.

**Comment by dperpeet** [comment]
needs rebase

File: pkg/registry/views/filter-bar.html [hunk 1]
```diff
@@ -1,3 +1,21 @@
+<div class="btn-group filter-menu pull-right" dropdown>
+    <button class="btn btn-default fa fa-th-list"></button>
+    <button class="btn btn-default dropdown-toggle" data-toggle="dropdown" dropdown-toggle>
+        <span class="caret"></span>
+    </button>
+    <ul class="dropdown-menu">
+        <li ng-class="{ active: listing.expandable }">
+            <a ng-click="listing.expandable = !listing.exandable">Expand items inline</a>
```
Comment (petervo) [inline pkg/registry/views/filter-bar.html]: Translateable

File: pkg/registry/views/filter-bar.html [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+        </li>
+        <li ng-class="{ active: listing.only }">
+            <a ng-click="listing.only = !listing.only">Only expanded items</a>
```
Comment (petervo) [inline pkg/registry/views/filter-bar.html]: Translateable

File: pkg/registry/views/filter-bar.html [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+        </li>
+        <li>
+            <a ng-click="listing.collapse()">Collapse all</a>
```
Comment (petervo) [inline pkg/registry/views/filter-bar.html]: Translateable

File: pkg/registry/views/image-page.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<div class="content-filter">
+    <div class="listing-actions">
+        <button class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
+    </div>
+    <h3>
+        <i class="pficon pficon-image"></i>
+        {{ imagestream.metadata.namespace }}/{{ imagestream.metadata.name}}:{{ tag.tag }}
+    </h3>
+    <a ng-href="#/images/{{ imagestream.metadata.namespace }}/{{ imagestream.metadata.name }}">Show all images</a>
```
Comment (petervo) [inline pkg/registry/views/image-page.html]: Translateable

File: pkg/registry/views/imagestream-page.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<div class="content-filter">
+    <div class="listing-actions">
+        <button class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
+    </div>
+    <h3>
+        <i class="fa fa-folder"></i>
+        {{ imagestream.metadata.namespace }}/{{ imagestream.metadata.name}}
+    </h3>
+    <a ng-href="{{ viewUrl('images') }}">Show all image streams</a>
```
Comment (petervo) [inline pkg/registry/views/imagestream-page.html]: Translateable

**Comment by petervo** [comment]
Mostly seemed to work well, Once I selected "expand items inline" i wasn't able to unselect it.

There are also a few new console.log in the tests, were those intentional?

**Comment by dperpeet** [comment]
> Mostly seemed to work well, Once I selected "expand items inline" i wasn't able to unselect it.

I think the minimal version of this should at least have the counterpart to "expand items inline", e.g. "open item on new page"

The other two options (that don't do anything) could maybe be hidden.

Also, I didn't notice the dropdown until you told me about it. On a wide screen it's very far away from the content. @andreasn What do you think?

**Comment by petervo** [comment]
FWIW, both other options work for me

**Comment by dperpeet** [comment]
> FWIW, both other options work for me

Ah, my bad. The collapse all items works. The filter `only expanded items` doesn't do anything.

**Comment by dperpeet** [comment]
> I think the minimal version of this should at least have the counterpart to "expand items inline", e.g. "open item on new page"

This could also be achieved via "checking" (highlighting?) the inline option.

File: pkg/registry/views/filter-bar.html (hunk 1 above)
Comment (stefwalter) [inline pkg/registry/views/filter-bar.html]: Added.

File: pkg/registry/views/filter-bar.html (hunk 2 above)
Comment (stefwalter) [inline pkg/registry/views/filter-bar.html]: Added.

File: pkg/registry/views/filter-bar.html (hunk 3 above)
Comment (stefwalter) [inline pkg/registry/views/filter-bar.html]: Added.

File: pkg/registry/views/image-page.html (hunk 4 above)
Comment (stefwalter) [inline pkg/registry/views/image-page.html]: Added

File: pkg/registry/views/imagestream-page.html (hunk 5 above)
Comment (stefwalter) [inline pkg/registry/views/imagestream-page.html]: Added.

**Comment by stefwalter** [comment]
> There are also a few new console.log in the tests, were those intentional?

Removed

**Comment by stefwalter** [comment]
Added check for selected menu items. Disabled items that don't make sense.

**Comment by dperpeet** [comment]
FileError: 'https://192.168.122.243:9090/cockpit/$c10142268eaa384a7949263743d9e4f75e676f76/registry/styles/dropdown.less' wasn't found (404)

**Comment by stefwalter** [comment]
Fixed and merged #3453 into this.

**Comment by stefwalter** [comment]
> Also, I didn't notice the dropdown until you told me about it. On a wide screen it's very far away from the content. @andreasn What do you think?

By the way this is a minor aspect of the UI, for advanced users, I'm fine with having it on the side on wide screens.

**Pull request closed without merging**